### PR TITLE
setting DNS resolution to rotate DNS servers

### PR DESCRIPTION
**Pull request by balsmn**
Fixes the issue reported in https://github.com/eclipse/hono/issues/332 for Hono Example Receiver and Hono JMeter Plugin

Signed-off-by: Balasubramanian <balasubramanian.azhagappan@bosch-si.com>

### CODE REVIEW

File: jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java [hunk 1]
```diff
@@ -166,4 +171,19 @@ public void close() {
             LOGGER.error("unknown exception in closing of receiver", t);
         }
     }
+
+    private Vertx vertx() {
+        VertxOptions options = new VertxOptions()
+                .setWarningExceptionTime(1500000000)
+                .setAddressResolverOptions(new AddressResolverOptions()
+                        .setCacheNegativeTimeToLive(0) // discard failed DNS lookup results immediately
+                        .setCacheMaxTimeToLive(0) // support DNS based service resolution
```
Comment (ppatierno) [inline jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java]: It's not strictly related to this PR but maybe we should avoid having these values hard coded and defining constants for them as well. Wdyt @sysexcontrol ?

Comment (sysexcontrol) [reply jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java]: @ppatierno : I fully agree - short investigation showed that we currently have the whole code block 6 times (!) inside of Hono.
I rather would thus propose a central location (util-class e.g.)  where this code block can be called:

`        VertxOptions options = new VertxOptions()
                .setWarningExceptionTime(1500000000)
                .setAddressResolverOptions(new AddressResolverOptions()
                        .setCacheNegativeTimeToLive(0) // discard failed DNS lookup results 
immediately
                        .setCacheMaxTimeToLive(0) // support DNS based service resolution
                        .setQueryTimeout(1000));
`

And extract these values to constants additionally.
WDYT?

@balsmn : this does not affect this PR from you, of course, and will be handled separately.

Comment (ppatierno) [reply jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java]: @sysexcontrol agree

File: jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java [hunk 2]
```diff
@@ -40,9 +42,12 @@
 
     private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(HonoReceiver.class);
 
+    private static final int DEFAULT_CONNECT_TIMEOUT_MILLIS = 2000;
+    private static final int DEFAULT_DNS_QUERY_TIMEOUT_MILLIS = 2000;
```
[the remark this answers is not in the document]
Comment (balsmn) [reply jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java]: But that means we will not have different timeouts for DNS resolution and connection timeout if it is ok.

File: jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java (hunk 1 above)
Comment (balsmn) [reply jmeter/src/main/java/org/eclipse/hono/jmeter/client/HonoReceiver.java]: That would be good.

**Pull request merged**